sourmash_lib/__main__.py

```diff
@@ -294,13 +297,26 @@ def save_siglist(siglist, output_fp, filename=None):
                 if args.singleton:
                     siglist = []
                     for n, record in enumerate(screed.open(filename)):
+                        if n % args.progress_stepsize == 0:
+                            if sys.version_info[0] < 3:
```

Comment:
Hmm, I think we should avoid adding too many of these checks to the code. Is it too bad to just not flush the output?

It is a progress report. Not flushing the output kind of defeats the purpose.

    And not for this PR, but we should definitely start using the logging module for this kind of output, instead of printing stuff. Also makes it easier to filter what output we want (INFO for printing progress, for example?)

Big yes about using logging (I am trying to match the current code base and not have changes and new features are the same time to the extent possible - I am happy to go both if not creating too much trouble). Note: Flushing is implicitly done by logging with each call, I believe.